conn.go

```diff
@@ -554,13 +558,37 @@ func Validate(request Message, replies []Message) error {
 	return nil
 }
 
+const (
+	// CallingThreadNetNS is a special value for Config.NetNS that allows Conn
+	// to enter the same Linux network namespace as the calling thread. Setting
+	// network namespaces is a privileged operation, so this option must be
+	// explicitly specified in order to do so.
+	//
+	// Note that the Go runtime multiplexes many goroutines onto many OS threads,
+	// and special care must be taken when manipulating thread state such as
+	// network namespaces. See the documentation of runtime.LockOSThread and
+	// runtime.UnlockOSThread for details.
+	//
+	// This option is useful in very specific circumstances, and must be used
+	// with great care.
+	CallingThreadNetNS = -1
```

Comment:
Name is up for discussion, but this felt most explicit. As for the value, I figure any negative number will do fine.